plumbing/reference.go

```diff
@@ -95,6 +95,14 @@ func (r ReferenceName) IsBranch() bool {
 	return strings.HasPrefix(string(r), refHeadPrefix)
 }
 
+// BranchName returns the branch name if the reference is a branch
+func (r ReferenceName) BranchName() string {
+	if r.IsBranch() {
+		return strings.TrimPrefix(string(r), refHeadPrefix)
+	}
+	return ""
+}
+
```

Comment:
We can't add both `BranchName()` and `TagName()` funcs to the ReferenceName, as that would weaken the abstraction. Neither is needed as per comment below.